Implementation of hdsortsortingextractore

rdiggelmann opened the pull request.

alejoe91 commented:
Thank you @rdiggelmann 

Waiting for the final changes as discussed before merging :)

samuelgarcia commented:
Hi @rdiggelmann.
In python, generally, upper case variable are used for class and not for  instance variable.
In your implementation you use `MultiElectrode` as a variable, I think you should go for lower case.
It would enhance readability for old pythonistas like me :)

alejoe91 commented:
> Hi @rdiggelmann.
> In python, generally, upper case variable are used for class and not for instance variable.
> In your implementation you use `MultiElectrode` as a variable, I think you should go for lower case.
> It would enhance readability for old pythonistas like me :)

@rdiggelmann told ya ;)

rdiggelmann commented:
Hi!
Yes, I'm still implementing the changes. I also added a unit test function but that one keeps failing. I'll commit as soon as I've fixed the problem.

@samuelgarcia Alessio explained me yesterday what style I should use. I also changed that and will commit all of it soon.

rdiggelmann commented:
I committed my changes and the tests should work now. I also tested it locally with a real dataset but I couldn't integrate that into a unittest.

alejoe91 commented on spikeextractors/extractors/matsortingextractor/matsortingextractor.py:
```diff
@@ -67,3 +75,6 @@ def _drill(d: dict, keys: deque):
             return self._data[fieldname][()]
 
+    @staticmethod
+    def write_dict_v7_3(mat_file_path, dict_to_write): # field must be a dict
```
would it make sense to have a `write_dict_to_mat()` function with a `version` argument? 

If >= 7.3 --> hdf5storage, otherwise scipy.io.savemat

@rdiggelmann what do you think?

alejoe91 commented on spikeextractors/extractors/hdsortsortingextractor/hdsortsortingextractor.py:
```diff
@@ -0,0 +1,179 @@
+                unit["spikeAmplitudes"] = sorting.get_unit_spike_features(uid, "amplitudes")
+            else:
+                # Save a spikeAmplitudes = 1
+                print("asdf")
```
could you remove this? ehehehe

alejoe91 commented:
@rdiggelmann still issues when writing and loading back regarding the shapes of the fooprints. Let's check it out on Monday ;)

alejoe91 commented on spikeextractors/extractors/jrcsortingextractor/jrcsortingextractor.py:
```diff
@@ -32,7 +29,7 @@ def __init__(self, file_path: PathType, keep_good_only: bool = False):
 
         # try to extract various parameters from the .prm file
-        self._kwargs["bit_scaling"] = np.float32(0.30518)  # conversion factor for ADC units -> µV
+        self._bit_scaling = np.float32(0.30518)  # conversion factor for ADC units -> µV
```
this is not a `self._kwargs`, which are only args of the `__init__`

alejoe91 commented on spikeextractors/extractors/matsortingextractor/matsortingextractor.py:
```diff
@@ -45,25 +54,37 @@ def __init__(self, file_path: PathType):
         try:  # load old-style (up to 7.2) .mat file
             self._data = loadmat(file_path, matlab_compatible=True)
-            self._kwargs["old_style_mat"] = True
+            self._old_style_mat = True
```
same here,not a `self._kwargs`

alejoe91 commented on spikeextractors/extractors/waveclussortingextractor/waveclussortingextractor.py:
```diff
@@ -16,7 +16,7 @@ class WaveClusSortingExtractor(MATSortingExtractor):
     installation_mesg = ""  # error message when not installed
 
-    def __init__(self, file_path: PathType, keep_good_only: bool = False):
+    def __init__(self, file_path: PathType):
```
`keep_good_only` was not used

alejoe91 commented on spikeextractors/tests/utils.py:
```diff
@@ -37,6 +38,13 @@ def check_recordings_equal(RX1, RX2):
 def check_recording_properties(RX1, RX2):
     # check properties
     assert sorted(RX1.get_shared_channel_property_names()) == sorted(RX2.get_shared_channel_property_names())
+    for prop in RX1.get_shared_channel_property_names():
```
added tests to compare saved and loaded properties and spike features
